File: Project37/backend/src/main/java/com/docarchive/controller/AuthController.java
package com.docarchive.controller;

import com.docarchive.common.Result;
import com.docarchive.dto.LoginRequest;
import com.docarchive.entity.User;
import com.docarchive.security.JwtTokenProvider;
import com.docarchive.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/auth")
public class AuthController {

    @Autowired
    private AuthenticationManager authenticationManager;

    @Autowired
    private JwtTokenProvider tokenProvider;

    @Autowired
    private UserService userService;

    @PostMapping("/login")
    public Result<Map<String, Object>> login(@RequestBody LoginRequest request) {
        Authentication authentication = authenticationManager.authenticate(
                new UsernamePasswordAuthenticationToken(request.getUsername(), request.getPassword())
        );

        SecurityContextHolder.getContext().setAuthentication(authentication);
        String jwt = tokenProvider.generateToken(authentication);

        User user = userService.findByUsername(request.getUsername()).orElse(null);

        Map<String, Object> data = new HashMap<>();
        data.put("token", jwt);
        data.put("username", user.getUsername());
        data.put("realName", user.getRealName());
        data.put("role", user.getRole());
        data.put("email", user.getEmail());

        return Result.success("登录成功", data);
    }

    @GetMapping("/me")
    public Result<User> getCurrentUser(Authentication authentication) {
        if (authentication == null) {
            return Result.error("未登录");
        }
        String username = authentication.getName();
        return userService.findByUsername(username)
                .map(Result::success)
                .orElse(Result.error("用户不存在"));
    }
}

File: Project29/backend/src/auth/auth.controller.ts
import { Controller, Post, Body, UseGuards, Get, Request, UnauthorizedException } from '@nestjs/common';
import { AuthService } from './auth.service';
import { AuthGuard } from '@nestjs/passport';

@Controller('api/auth')
export class AuthController {
  constructor(private authService: AuthService) {}

  @Post('register')
  async register(@Body() body: { username: string; email: string; password: string }) {
    const user = await this.authService.register(body.username, body.email, body.password);
    const { password, ...result } = user;
    return { success: true, data: result };
  }

  @Post('login')
  async login(@Body() body: { username: string; password: string }) {
    const result = await this.authService.login(body.username, body.password);
    return { success: true, data: result };
  }

  @Get('profile')
  @UseGuards(AuthGuard('jwt'))
  async getProfile(@Request() req) {
    const user = await this.authService.findById(req.user.userId);
    if (!user) {
      throw new UnauthorizedException('User not found');
    }
    const { password, ...result } = user;
    return { success: true, data: result };
  }
}
